core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -791,7 +790,7 @@ class UnifiedLog(@volatile var logStartOffset: Long,
             val offset = PrimitiveRef.ofLong(localLog.logEndOffset)
             appendInfo.setFirstOffset(offset.value)
             val validateAndOffsetAssignResult = try {
-              val targetCompression = BrokerCompressionType.forName(config.compressionType).targetCompressionType(appendInfo.sourceCompression)
+              val targetCompression = config.compressionType.targetCompression(config.compression, appendInfo.sourceCompression())
```

Comment:
`config.compressionType` is not really being used. Could we make `targetCompression` a static method and independent of compressionType?